### PR TITLE
#12 angular class generation options

### PR DESCRIPTION
**Pull request by ZimGil**
Fully implemented basic options for all supported `ng` commands

solves #12

### CODE REVIEW

File: src/app/components/generate/generate.component.html [hunk 1]
```diff
@@ -1,12 +1,34 @@
...
 </select>
 
-<button (click)="generateAngularClass()" [disabled]="!name">Generate</button>
+<div *ngIf="options.mandatoryArgs.type == angularClass.component">
+  <label>
+    <input type="checkbox" [(ngModel)]="options.optionalFlags.inLineTemplate.isActive">
+    <span>in-line template</span>
+  </label>
+</div>
+<div *ngIf="options.mandatoryArgs.type == angularClass.component">
```
Comment (ortichon) [inline src/app/components/generate/generate.component.html]: we don't have to use `ngIf` twice. just use it once on a common parent

File: src/app/components/generate/generate.component.ts [hunk 2]
```diff
@@ -1,34 +1,51 @@
-import { Component, Output, EventEmitter } from '@angular/core';
+import { Component, Output, EventEmitter, OnInit } from '@angular/core';
 
 import { AngularClass } from '../../models/angular-class.enum';
 import { AngularClassOptions } from '../../models/angular-class-options.interface';
 import { AngularCliCommand } from '../../models/angular-cli-command.interface';
+import { NgOptions } from './../../models/angular-options';
```
Comment (ortichon) [inline src/app/components/generate/generate.component.ts]: remove `./`

File: src/app/components/generate/generate.component.ts [hunk 3]
```diff
@@ -1,34 +1,51 @@
...
-    this.sendCommand.emit(this.command);
+    this.sendCommand.emit(this.options.createCommandString());
+  }
+
+  changeType(type: AngularClass): void {
+    this.previousName = <string>this.options.mandatoryArgs.name;
+    this.options = this.NgOptionsFactory(type);
+  }
+
+  NgOptionsFactory(type: AngularClass): NgOptions {
```
Comment (ortichon) [inline src/app/components/generate/generate.component.ts]: methods should be lowered-case

File: src/app/components/new/new.component.html [hunk 4]
```diff
@@ -1,4 +1,24 @@
 <h4>Create a New Project</h4>
-<input type="text" [(ngModel)]="name">
+<input type="text" [(ngModel)]="options.mandatoryArgs.name">
 
+<div>
+<label>
```
Comment (ortichon) [inline src/app/components/new/new.component.html]: indentation

File: src/app/components/new/new.component.html [hunk 5]
```diff
@@ -1,4 +1,24 @@
...
+    <input type="checkbox" [(ngModel)]="options.optionalFlags.inLineTemplate.isActive">
+    <span>in-line template</span>
+</label>
+</div>
+
+<div>
```
Comment (ortichon) [inline src/app/components/new/new.component.html]: we don't need this `div`

File: src/app/components/new/new.component.html [hunk 6]
```diff
@@ -1,4 +1,24 @@
...
+    <label>
+        <input type="checkbox" [(ngModel)]="options.optionalFlags.inLineStyle.isActive">
+        <span>in-line style</span>
+    </label>
+</div>
+
+<div>
```
Comment (ortichon) [inline src/app/components/new/new.component.html]: we don't need this `div`

File: src/app/components/serve/serve.component.html [hunk 7]
```diff
@@ -1,9 +1,40 @@
 <div>Hostname / IP</div>
 
-<input type="text" [(ngModel)]="hostname">
-:
-<input type="text" [(ngModel)]="port">
-<input type="checkbox" [(ngModel)]="isOpen">
+<div>
```
Comment (ortichon) [inline src/app/components/serve/serve.component.html]: we don't need this `div`

File: src/app/components/serve/serve.component.html [hunk 8]
```diff
@@ -1,9 +1,40 @@
...
+            [(ngModel)]="options.optionalFlags.host.isActive">
+        <span>Add Hoatname </span>
+        <input 
+        *ngIf="options.optionalFlags.host.isActive"
+        type="text" 
+        [(ngModel)]="options.optionalFlags.host.params">
+    </label>
+</div>
 
+<div>
```
Comment (ortichon) [inline src/app/components/serve/serve.component.html]: redundant `div`

File: src/app/components/serve/serve.component.html [hunk 9]
```diff
@@ -1,9 +1,40 @@
...
+            type="checkbox"
+            [(ngModel)]="options.optionalFlags.port.isActive">
+        <span>Add Port </span>
+        <input
+            *ngIf="options.optionalFlags.port.isActive"
+            type="text"
+            [(ngModel)]="options.optionalFlags.port.params">
+    </label>
+</div>
+<div>
```
Comment (ortichon) [inline src/app/components/serve/serve.component.html]: redundant

File: src/app/components/serve/serve.component.ts [hunk 10]
```diff
@@ -1,6 +1,7 @@
 import { Component, Output, EventEmitter } from '@angular/core';
 
 import { AngularCliCommand } from '../../models/angular-cli-command.interface';
+import { NgserveOptions } from './../../default-values/ng-serve-options';
```
Comment (ortichon) [inline src/app/components/serve/serve.component.ts]: remove `./`

File: src/app/components/serve/serve.component.ts [hunk 11]
```diff
@@ -12,30 +13,21 @@ export class ServeComponent {
   private readonly MINIMUM_PORT = 1000;
   private readonly MAXIMUM_PORT = 99999;
 
-  @Output() sendCommand = new EventEmitter<AngularCliCommand>();
+  @Output() sendCommand = new EventEmitter<string>();
   command: AngularCliCommand;
-  hostname = "localhost";
-  port = 4200;
-  isOpen = true;
+  options = new NgserveOptions
```
Comment (ortichon) [inline src/app/components/serve/serve.component.ts]: use `()` when creating a new class

File: src/app/default-values/ng-generate-component-options.ts [hunk 12]
```diff
@@ -0,0 +1,48 @@
+import { NgOptions } from './../models/angular-options';
```
Comment (ortichon) [inline src/app/default-values/ng-generate-component-options.ts]: `./`

File: src/app/components/serve/serve.component.ts [hunk 13]
```diff
@@ -12,30 +13,21 @@ export class ServeComponent {
...
-        {isOpen: this.isOpen}
-      ]
-    }
-    this.sendCommand.emit(this.command);
+    this.sendCommand.emit(this.options.createCommandString());
   }
 
   isPortValid(): boolean {
-    return this.port >= this.MINIMUM_PORT && this.port <= this.MAXIMUM_PORT;
+    return this.options.optionalFlags.port.params >= this.MINIMUM_PORT
```
Comment (ortichon) [inline src/app/components/serve/serve.component.ts]: looks better:
```typescript
const port = this.options.optionalFlags.port.params;
return port >= this.MINIMUM_PORT && port <= this.MAXIMUM_PORT;
```

File: src/app/default-values/ng-serve-options.ts [hunk 14]
```diff
@@ -0,0 +1,43 @@
...
+            if (option.isActive) {
+                command.push(option.flag);
+                if (option.params) {
+                    command.push(option.params.toString());
+                }
+            }  
+        })
+        return command.join(' ');
+    }
+}
```
Comment (ortichon) [inline src/app/default-values/ng-serve-options.ts]: EOF

**Comment by ZimGil** [comment]
All issues have been addressed...
Please review @ortichon

**Comment by ZimGil** [comment]
@ortichon Please wait for fixing of issue #28

File: src/app/components/generate/generate.component.html (hunk 1 above)
Comment (ZimGil) [reply src/app/components/generate/generate.component.html]: @Thatkookooguy 
Thanks, I've looked into it, very cool.
I guess it's not necessary now as we only have 2 check-boxes in the same div.

**Comment by ZimGil** [comment]
#28 resolved.
@ortichon please review

File: .vscode/settings.json [hunk 15]
```diff
@@ -0,0 +1,2 @@
+{
```
Comment (ortichon) [inline .vscode/settings.json]: please add `.vscode` to the `.gitignore` file in order to ignore prevent pushing files like this one to the source

**Comment by ZimGil** [comment]
`.gitignore` was updated.
@ortichon

File: src/app/models/angular-options.ts [hunk 16]
```diff
@@ -0,0 +1,10 @@
+import { AngularCommandOptions } from './angular-command-options.interface';
+
+export interface NgOptions {
+    mandatoryArgs: {
+        [argName: string]: number | string;
+    }
+    optionalFlags: AngularCommandOptions;
+
+    createCommandString(): string;
+}
```
Comment (ortichon) [inline src/app/models/angular-options.ts]: EOF

**Pull request merged**